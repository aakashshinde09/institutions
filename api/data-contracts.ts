/* eslint-disable */
/* tslint:disable */
/*
 * ---------------------------------------------------------------
 * ## THIS FILE WAS GENERATED VIA SWAGGER-TYPESCRIPT-API        ##
 * ##                                                           ##
 * ## AUTHOR: acacode                                           ##
 * ## SOURCE: https://github.com/acacode/swagger-typescript-api ##
 * ---------------------------------------------------------------
 */

export interface NGOLookup {
  id: number
  graphId: string
  name: string
  imageLink: string
  description: string
  link: string
  rewardsOwner: string
  ngoAddress: string
  totalStacked: number
  totalDonation: number
  lastDonation?: string
  donationPercent?: number
  totalUserStaked: number
  location: string
  balance: number
  stakes?: Stakes[]
}

export interface Stakes {
  balance: number
  createdAt: string
  customStakeId: number
  stackedAmount: number
  percentShare: number
  totalDonation: number
  startDate: string
}

export type ApiResponse = object

export interface RewardLookup {
  id: number
  stackedAmount: number
  balanceWithoutProfit: number
  profit: string
  donation: string
  ngoAddress: string
  /** @format date-time */
  createdAt: string
  serviceFee: string
  stake?: Stakes
}

export interface AuthTokensDto {
  accessToken: string
  refreshToken: string
}

export interface SignDto {
  publicAddress: string
}

export interface AprDto {
  Apr: number
  Timestamp: number
}

export interface AverageAprDto {
  Aprs: AprDto[]
  SmaArp: number
  Symbol: string
  Address: string
  ChainId: number
}

export interface TransactionLookup {
  id: number
  hash: string
  transactionType: string
  fromAddress: string
  toAddress: string
  ethAmount: number
  networkFee: number
  blockNumber: number
  blockTimestamp: number
}

export interface WithdrawRequestLookup {
  id: number
  graphId: string
  customRequestId: number
  ngoAddress: string
  status: 1 | 2
  stakerAddress: string
  createdAt: string
  stake: Stakes
}

export interface WithdrawStEthRequestLookup {
  id: string
  ngo: string
  claimer: string
  stake: Stakes
  blockTimestamp: number
}

export interface NgoControllerGetAllParams {
  take: number
  skip: number
}

export type NgoControllerGetAllData = ApiResponse & {
  data?: {
    items?: NGOLookup[]
    /** @default 1 */
    totalItems?: number
  }
  /** @default 200 */
  status?: number
  error?: string
}

export type NgoControllerPullNgoFromGraphData = any

export type NgoControllerDeleteNgoByIdData = any

export interface RewardControllerGetCurrentUserRewardsParams {
  take: number
  skip: number
}

export type RewardControllerGetCurrentUserRewardsData = ApiResponse & {
  data?: {
    items?: RewardLookup[]
    /** @default 1 */
    totalItems?: number
  }
  /** @default 200 */
  status?: number
  error?: string
}

export type RewardControllerPullRewardFromGraphAndScData = any

export type RewardControllerDeleteAllData = any

export type RewardControllerDeleteOneData = any

export type UserControllerGeCurrentUserDepositsInformationData = any

export type UserControllerGetCurrentUserNgoListData = ApiResponse & {
  data: NGOLookup[]
  /** @default 200 */
  status?: number
  error?: string
}

export type AuthControllerSignInData = ApiResponse & {
  data?: AuthTokensDto
  /** @default 200 */
  status?: number
  error?: string
}

export type AuthControllerSignUpData = any

export interface AuthControllerRefreshPayload {
  refreshToken?: string
}

export type AuthControllerRefreshData = ApiResponse & {
  data?: AuthTokensDto
  /** @default 200 */
  status?: number
  error?: string
}

export type AprControllerGetAverageAprData = ApiResponse & {
  data?: AverageAprDto
  /** @default 200 */
  status?: number
  error?: string
}

export interface TransactionControllerGetByWalletAddressParams {
  take: number
  skip: number
}

export type TransactionControllerGetByWalletAddressData = ApiResponse & {
  data?: {
    items?: TransactionLookup[]
    /** @default 1 */
    totalItems?: number
  }
  /** @default 200 */
  status?: number
  error?: string
}

export interface WithdrawControllerGetByWalletAddressParams {
  take: number
  skip: number
  status: 1 | 2
}

export interface withdrawControllerGetWithdrawStEth {
  take: number
  skip: number
  claimerAddress: string
}

export type WithdrawControllerGetByWalletAddressData<T = WithdrawRequestLookup> =
  ApiResponse & {
    data?: {
      items?: T[]
      /** @default 1 */
      totalItems?: number
    }
    /** @default 200 */
    status?: number
    error?: string
  }
