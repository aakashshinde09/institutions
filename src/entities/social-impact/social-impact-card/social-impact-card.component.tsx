import type { NGOLookup } from '@api/data-contracts'
import { elementIds } from '@shared/lib/enums'
import { formatNumber, scannerLink } from '@shared/lib/utils'
import { Button } from '@shared/ui/button'
import { IconComponent } from '@shared/ui/icon'
import Image from 'next/image'

interface ISocialImpactCardProperties {
  data: NGOLookup
  ind: number
}

export const SocialImpactCard = ({ data, ind }: ISocialImpactCardProperties) => {
  return (
    <div className="flex h-full flex-col rounded-3xl bg-white p-3 md:pb-4">
      <div className="relative h-[18.75rem] w-full overflow-hidden rounded-[1.25rem] max-md:h-[15rem]">
        <Image
          src={
            data.imageLink.includes('http') ? data.imageLink : `https://${data.imageLink}`
          }
          alt="Social impact image"
          sizes="25rem, 18.75rem"
          fill
          className="object-scale-down"
          priority
        />
        <div className="absolute right-3 top-3 flex items-center gap-3">
          <a
            href={scannerLink('address', data.ngoAddress)}
            target="_blank"
            rel="noreferrer"
            className="flex h-12 w-12 items-center justify-center rounded-full border-none bg-white p-3"
          >
            <IconComponent name="document" viewBox="0 0 24 24" className="text-primary" />
          </a>
          <a
            href={data.link}
            target="_blank"
            rel="noreferrer"
            className="flex h-12 w-12 items-center justify-center rounded-full border-none bg-white p-3"
          >
            <IconComponent name="web" viewBox="0 0 24 24" />
          </a>
        </div>
      </div>

      <div className="mb-5 mt-6 max-md:mt-4">
        <h3 className="h3">{data.name}</h3>
        <div className="mt-2 flex items-center gap-1 max-md:text-sm">
          <IconComponent name="office" viewBox="0 0 24 24" className="h-6 w-6" />
          <span className="label-2 text-black/80">{data.location}</span>
        </div>
        <div className='max-h-[120px] overflow-auto scroll pr-2 mt-2'><p className="mt-3 text-base text-muted max-md:text-sm">{data.description}</p></div>
      </div>

      <div className="mt-auto flex items-center justify-between rounded-sl bg-primary/20 pr-3 max-md:mt-4">
        <Button
          id={elementIds.STAKE_N_DONATE + ind}
          className="max-md:py-4"
          href={`/staking?ngo=${data.id}`}
        >
          stake&donate
        </Button>
        <div className="ml-1 flex grow items-center justify-center gap-1 text-base max-md:text-sm">
          <span className="text-primary/50">Total staked:</span>
          <span className="font-semibold text-primary">
            {formatNumber(data.totalStacked)} stETH
          </span>
        </div>
      </div>
    </div>
  )
}
