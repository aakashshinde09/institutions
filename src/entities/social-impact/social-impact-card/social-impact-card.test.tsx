import { expect, test, describe } from 'vitest'
import { render } from '@testing-library/react'
import { SocialImpactCard } from './social-impact-card.component'
import { NGOLookup } from '@api/data-contracts'

const mock: NGOLookup = {
  id: 1,
  graphId: '0xf5043d3e6ee432ee3fb8e10b81a05e3e261552aa8c621230ed6457125fe6bb69',
  name: 'UNICEF',
  imageLink: 'https://i.pinimg.com/736x/50/ba/9f/50ba9f98aaf8f8ded4d576a6969668f0.jpg',
  description:
    "UNICEF works in over 190 countries and territories to save children's lives to defend their rights and to help them fulfil their potential from early childhood through adolescence. And we never give up.",
  link: 'https://www.unicef.org/',
  totalStacked: 0.03639999999999999,
  location: 'United Kingdom',
  ngoAddress: '0x30de5ffc4be9864506336bc908481470611a6d93',
  rewardsOwner: '0xdf9c8efb80f2dad58d2e2d7c30f1d190c02197a8',
  totalUserStaked: 4,
  balance: 100,
  totalDonation: 1000
}

describe('<SocialImpactCard/>', () => {
  test('Renders correctly', () => {
    const component = render(<SocialImpactCard data={mock} ind={1} />)

    expect(component).toMatchSnapshot()
  })
})
