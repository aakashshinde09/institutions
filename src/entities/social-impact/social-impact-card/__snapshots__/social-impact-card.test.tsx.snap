// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SocialImpactCard/> > Renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex h-full flex-col rounded-3xl bg-white p-3 md:pb-4"
      >
        <div
          class="relative h-[18.75rem] w-full overflow-hidden rounded-[1.25rem] max-md:h-[15rem]"
        >
          <img
            alt="Social impact image"
            class="object-scale-down"
            data-nimg="fill"
            decoding="async"
            fetchpriority="high"
            sizes="25rem, 18.75rem"
            src="/_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=3840&q=75"
            srcset="/_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=3840&q=75 3840w"
            style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
          />
          <div
            class="absolute right-3 top-3 flex items-center gap-3"
          >
            <a
              class="flex h-12 w-12 items-center justify-center rounded-full border-none bg-white p-3"
              href="https://holesky.etherscan.io/address/0x30de5ffc4be9864506336bc908481470611a6d93"
              rel="noreferrer"
              target="_blank"
            />
            <a
              class="flex h-12 w-12 items-center justify-center rounded-full border-none bg-white p-3"
              href="https://www.unicef.org/"
              rel="noreferrer"
              target="_blank"
            />
          </div>
        </div>
        <div
          class="mb-5 mt-6 max-md:mt-4"
        >
          <h3
            class="h3"
          >
            UNICEF
          </h3>
          <div
            class="mt-2 flex items-center gap-1 max-md:text-sm"
          >
            <span
              class="label-2 text-black/80"
            >
              United Kingdom
            </span>
          </div>
          <p
            class="mt-3 text-base text-muted max-md:text-sm"
          >
            UNICEF works in over 190 countries and territories to save children's lives to defend their rights and to help them fulfil their potential from early childhood through adolescence. And we never give up.
          </p>
        </div>
        <div
          class="mt-auto flex items-center justify-between rounded-sl bg-primary/20 pr-3 max-md:mt-4"
        >
          <a
            class="rounded-sl px-12 py-5 text-base transition-colors cursor-pointer font-semibold uppercase no-underline max-md:px-6 max-md:text-sm disabled:cursor-not-allowed bg-primary text-white border-none hover:bg-primary/90 disabled:bg-primary/70 max-md:py-4"
            href="/staking?ngo=1"
            id="stakeNDonateBtn1"
          >
            stake&donate
          </a>
          <div
            class="ml-1 flex grow items-center justify-center gap-1 text-base max-md:text-sm"
          >
            <span
              class="text-primary/50"
            >
              Total staked:
            </span>
            <span
              class="font-semibold text-primary"
            >
              0.0364
               stETH
            </span>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex h-full flex-col rounded-3xl bg-white p-3 md:pb-4"
    >
      <div
        class="relative h-[18.75rem] w-full overflow-hidden rounded-[1.25rem] max-md:h-[15rem]"
      >
        <img
          alt="Social impact image"
          class="object-scale-down"
          data-nimg="fill"
          decoding="async"
          fetchpriority="high"
          sizes="25rem, 18.75rem"
          src="/_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=3840&q=75"
          srcset="/_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=16&q=75 16w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=32&q=75 32w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=48&q=75 48w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=64&q=75 64w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=96&q=75 96w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=128&q=75 128w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fi.pinimg.com%2F736x%2F50%2Fba%2F9f%2F50ba9f98aaf8f8ded4d576a6969668f0.jpg&w=3840&q=75 3840w"
          style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent;"
        />
        <div
          class="absolute right-3 top-3 flex items-center gap-3"
        >
          <a
            class="flex h-12 w-12 items-center justify-center rounded-full border-none bg-white p-3"
            href="https://holesky.etherscan.io/address/0x30de5ffc4be9864506336bc908481470611a6d93"
            rel="noreferrer"
            target="_blank"
          />
          <a
            class="flex h-12 w-12 items-center justify-center rounded-full border-none bg-white p-3"
            href="https://www.unicef.org/"
            rel="noreferrer"
            target="_blank"
          />
        </div>
      </div>
      <div
        class="mb-5 mt-6 max-md:mt-4"
      >
        <h3
          class="h3"
        >
          UNICEF
        </h3>
        <div
          class="mt-2 flex items-center gap-1 max-md:text-sm"
        >
          <span
            class="label-2 text-black/80"
          >
            United Kingdom
          </span>
        </div>
        <p
          class="mt-3 text-base text-muted max-md:text-sm"
        >
          UNICEF works in over 190 countries and territories to save children's lives to defend their rights and to help them fulfil their potential from early childhood through adolescence. And we never give up.
        </p>
      </div>
      <div
        class="mt-auto flex items-center justify-between rounded-sl bg-primary/20 pr-3 max-md:mt-4"
      >
        <a
          class="rounded-sl px-12 py-5 text-base transition-colors cursor-pointer font-semibold uppercase no-underline max-md:px-6 max-md:text-sm disabled:cursor-not-allowed bg-primary text-white border-none hover:bg-primary/90 disabled:bg-primary/70 max-md:py-4"
          href="/staking?ngo=1"
          id="stakeNDonateBtn1"
        >
          stake&donate
        </a>
        <div
          class="ml-1 flex grow items-center justify-center gap-1 text-base max-md:text-sm"
        >
          <span
            class="text-primary/50"
          >
            Total staked:
          </span>
          <span
            class="font-semibold text-primary"
          >
            0.0364
             stETH
          </span>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
